cloudfoundry/cf-create-org.sh

```diff
@@ -1,134 +1,124 @@
-HASHED_QUOTA_NAME=$(md5 -qs $QUOTA_NAME)
-
-# Step 0: Confirm the inputs.
-# http://stackoverflow.com/a/226724/358804
-cat << EOF
-Create the following?
-
-Quota: $HASHED_QUOTA_NAME
-Org: $ORG_NAME
-
-Enter 1 for "Yes" and 2 for "No"
-
-EOF
-select yn in "Yes" "No"; do
-  case $yn in
-    Yes ) break;;
-    No ) exit 1;;
-  esac
-done
-
-# Step 1: Check that we're logged into Concourse
 CI_URL="${CI_URL:-"https://ci.fr.cloud.gov"}"
 FLY_TARGET=$(fly targets | grep "${CI_URL}" | head -n 1 | awk '{print $1}')
 
-if ! fly --target "${FLY_TARGET}" workers > /dev/null; then
-  echo "Not logged in to concourse"
-  exit 1
-fi
-
-# Step 2: Create the quota
-cf create-quota "$HASHED_QUOTA_NAME" -m "$MEMORY" -r "$NUMBER_OF_ROUTES" -s "$NUMBER_OF_SERVICES" --allow-paid-service-plans
+function usage {
+cat >&2 <<EOM
+usage: $(basename "$0") -i iaa -o org_name -m manager[,origin] -m manager[,origin] -q quota_in_GB(8)
+
+examples:
+  $0 -i CFXXX250001-0002 -o dept-office-system -m pickle@gsa.gov,gsa.gov -m gherkin@dept.gov -q 8
+  $0 -i CFXXX250001-0001 -o dept-office-another -m okra@dept.gov -q 16
+EOM
+}
+
+function check_auth {
+  if ! fly --target "${FLY_TARGET}" workers > /dev/null; then
+    raise "Not logged in to Concourse: $CI_URL"
+  fi
+
+  cf oauth-token 1>/dev/null || raise "Not logged in to Cloud.gov's Cloud Foundry"
+}
+
+while getopts ":hi:m:o:q:" opt; do
+  case ${opt} in
+    h ) 
+      usage
+      exit 0 ;;
+    i )
+      IAA=$OPTARG;;
+    m )
+      MANAGERS+=("$OPTARG");;
+    o )
+      ORG_NAME=$OPTARG;;
+    q )
+      MEMORY_QUOTA=$OPTARG;;
+    \? ) 
+      raise "Invalid option: $OPTARG";;
+    : )
+      raise "Invalid option: $OPTARG requires an argument" ;;
+  esac
+done
+shift $((OPTIND-1))
 
-ADMIN=$(cf target | grep -i user | awk '{print $2}')
+# Make sure we're logged in, and check arguments
+check_auth 
 
-# Step 3: Create the org
-cf create-org "$ORG_NAME" -q "$HASHED_QUOTA_NAME"
-cf set-label org "$ORG_NAME" org-type=customer
+[[ -z $IAA ]] && ( usage ; raise "'-i iaa' is REQUIRED" )
+[[ -z $ORG_NAME  ]] && ( usage ; raise "'-o org_name' is REQUIRED" )
+[[ -z "${MANAGERS[0]}" ]] && ( usage ; raise "'-m manager_email' is REQUIRED at least once" )
 
-# creator added by default, which is usually not desirable
-cf unset-org-role "$ADMIN" "$ORG_NAME" OrgManager
-cf set-org-role "$MANAGER" "$ORG_NAME" OrgManager ${ORIGIN_FLAG:+"$ORIGIN_FLAG"}
+set -u # exit on unbound variables
+if ! [[ $ORG_NAME =~ ^[a-zA-Z0-9-]+$ ]]; then
+  raise "ORG_NAME must contain only letters, numbers, and hypens."
+fi
+if ! [[ $IAA =~ ^[a-zA-Z0-9_\.-]+$ ]]; then
+  raise "IAA must contain only letters, numbers, underscores, periods, and hyphens."
+fi
 
-# Step 4: Create the spaces
+# Add epoch time to avoid naming collisions
+NOW=$(date '+%s')
+# Hash the name to obfuscate the origin of the quota from snooping users
+QUOTA_NAME=$( echo "${ORG_NAME}_${IAA}_${NOW}" | tr '[:lower:]' '[:upper:]' | md5 -q )
```

Comment:
Is there any risk in included IAA numbers in quota names? Besides us, I guess customers can see their own quota names?